cluster-provision/gocli/docker/adapter.go

```diff
@@ -0,0 +1,48 @@
+package docker
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/docker/client"
+)
+
+// DockerAdapter is a wrapper around client.Client to conform it to the SSH interface
+type DockerAdapter struct {
+	nodeName     string
+	dockerClient *client.Client
+}
+
+func NewDockerAdapter(cli *client.Client, nodeName string) *DockerAdapter {
+	return &DockerAdapter{
+		nodeName:     nodeName,
+		dockerClient: cli,
+	}
+}
+
+func (d *DockerAdapter) SSH(cmd string) error {
```

Comment:
>Also there's refactoring potential, since I've seen it being used repeatedly inside a loop over a string slice, maybe it would make sense to change the func into variadic type

The first goal is to set abstraction and showcase how we can unit test this. I agree that there can be more done but we need to prioritize and go step by step